crates/ark/src/variables/variable.rs

```diff
@@ -90,8 +92,13 @@ impl WorkspaceVariableDisplayValue {
             },
             CLOSXP => Self::from_closure(value),
             ENVSXP => Self::from_env(value),
+            LANGSXP => Self::from_language(value),
             _ if r_is_matrix(value) => Self::from_matrix(value),
-            _ => Self::from_default(value),
+            RAWSXP | LGLSXP | INTSXP | REALSXP | STRSXP | CPLXSXP => Self::from_default(value),
+            _ => Self::from_error(Error::Anyhow(anyhow!(
+                "Unexpected type {}",
+                r_type2char(r_typeof(value))
+            ))),
```

Comment:
why this new restriction? For explicitness?

Reply:
yes, I taught that capturing an unexpected type here would be more explicit.
Before that, for example, the formula support error was being captured at: https://github.com/posit-dev/ark/blob/92ce2f73fdf71c06a34b1041fa5f22cf09a5efdc/crates/harp/src/vector/formatted_vector.rs#L112-L115

Which is fine, but, AFAICT we should only use `FormattedVector` for vector like objects.